### PR TITLE
HW-52500  Update Core Unit test to test isEditable field

### PR DESCRIPTION
**Pull request by ymoskovko-epam**
[Jira Ticket HW-52500](https://jira.site1.hyperwallet.local/browse/HW-52500)

### CODE REVIEW

File: Tests/TransferMethodConfigurationResultTests.swift [hunk 1]
```diff
@@ -194,12 +194,13 @@ class TransferMethodConfigurationResultTests: XCTestCase {
...
         XCTAssertEqual(fields.count, 3, "The amount of elements is different from the expected value")
         // Assert Field bankAccountPurpose
-        let fieldsFilterByBankAccountPurpose = fields.filter { (field) in  field.name == "bankAccountPurpose" }
+        let fieldsFilterByBankAccountPurpose = fields.filter { $0.name == "bankAccountPurpose" }
         if let bankAccountPurpose = fieldsFilterByBankAccountPurpose.first {
             XCTAssertEqual(bankAccountPurpose.name, "bankAccountPurpose", "Invalid name")
             XCTAssertEqual(bankAccountPurpose.category, "ACCOUNT", "Invalid category")
             XCTAssertEqual(bankAccountPurpose.dataType, "SELECTION", "Invalid dataType")
             XCTAssertEqual(bankAccountPurpose.isRequired, true, "Should be required")
+            XCTAssertNil(bankAccountPurpose.isEditable, "Should be nil")
```
Comment (syyang-hw) [inline Tests/TransferMethodConfigurationResultTests.swift]: ```nil``` or ```false```?

File: Tests/TransferMethodConfigurationResultTests.swift [hunk 2]
```diff
@@ -215,13 +216,13 @@ class TransferMethodConfigurationResultTests: XCTestCase {
...
         // Assert Field branchId
-        let fieldsFilterByBranchId = fields.filter { (field) in  field.name == "branchId" }
+        let fieldsFilterByBranchId = fields.filter { $0.name == "branchId" }
         if let branchId = fieldsFilterByBranchId.first {
             XCTAssertEqual(branchId.name, "branchId", "Invalid name")
             XCTAssertEqual(branchId.category, "ACCOUNT", "Invalid category")
             XCTAssertEqual(branchId.dataType, "NUMBER", "Invalid dataType")
-            XCTAssertEqual(branchId.isRequired, true, "Should not be required")
-            XCTAssertEqual(branchId.isEditable, true, "Should not be required")
+            XCTAssertEqual(branchId.isRequired, false, "Should not be required")
```
Comment (syyang-hw) [inline Tests/TransferMethodConfigurationResultTests.swift]: You could use ```XCAssertTrue```

File: Tests/Responses/TransferMethodConfigurationFieldsResponse.json [hunk 3]
```diff
@@ -27,7 +27,7 @@
                         {
                             "category": "ACCOUNT",
                             "dataType": "NUMBER",
-                            "isRequired": true,
+                            "isRequired": false,
```
Comment (jsaini-hw) [inline Tests/Responses/TransferMethodConfigurationFieldsResponse.json]: why did you change it to false? This field is always true

File: Tests/TransferMethodConfigurationResultTests.swift (hunk 1 above)
Comment (syyang-hw) [reply Tests/TransferMethodConfigurationResultTests.swift]: My bad, please ignore above comment

File: Tests/TransferMethodConfigurationResultTests.swift (hunk 2 above)
Comment (jsaini-hw) [reply Tests/TransferMethodConfigurationResultTests.swift]: You shouldn't change this. It's not part of the ticket. Ticket is about isEditable and not isRequired

File: Tests/Responses/TransferMethodConfigurationFieldsResponse.json (hunk 3 above)
Comment (ymoskovko-epam) [reply Tests/Responses/TransferMethodConfigurationFieldsResponse.json]: just for test

Comment (ymoskovko-epam) [reply Tests/Responses/TransferMethodConfigurationFieldsResponse.json]: it is fake data and we can check not only for true

File: Tests/TransferMethodConfigurationResultTests.swift (hunk 2 above)
Comment (ymoskovko-epam) [reply Tests/TransferMethodConfigurationResultTests.swift]: ok

File: Tests/Responses/TransferMethodConfigurationFieldsResponse.json (hunk 3 above)
Comment (ymoskovko-epam) [reply Tests/Responses/TransferMethodConfigurationFieldsResponse.json]: i will revert
